### PR TITLE
Ender Crytal Fix and report proper cmake file for luabindingscheck fail

### PR DESCRIPTION
**Pull request by 12xx12**
#5013 is stopping on nether generation again

### CODE REVIEW

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -9539,6 +9539,78 @@ a_Player:OpenWindow(Window);
...
+					Returns =
+					{
+						{
+							Type = "boolean",
+						},
+						Notes = "Returns true if the bedrock base is displayed.",
+					},
+				},
+			},
+		},
```
Comment (Seadragon91) [inline Server/Plugins/APIDump/APIDesc.lua:9613]: The class cEnderCrystal inherits from cEntity in c++-class. This required to have the entity functions in the api website: `Inherits = "cEntity",`

Comment (12xx12) [reply Server/Plugins/APIDump/APIDesc.lua:9614]: i'm confused by the error message generated by the checkstyle check. Any other class added in the ```src/Bindings/AllToLua.pkg``` are not in the ```CMakeList.txt``` file

Comment (tigerw) [reply Server/Plugins/APIDump/APIDesc.lua:9614]: The error message is outdated, try adding to CMake\GenerateBindings.cmake (by the way could you please update the message 😅)

**Comment by tigerw** [comment]
Hey 12xx12, I've prepared two small changes that I want to add;

1. Change order of members in cEnderCrystal (put all the bools after) for a small sizeof reduction (360B down to 352B)
2. Make cWorld::SpawnEnderCrystal only take a position and ShowBase, as you had before, just for easier usage for plugins. The reasoning being that the beam target is likely to constantly change, so plugins should favour the setter instead.

Does this sound ok to you?

**Comment by 12xx12** [comment]
Hi @tigerw 

1. yeah that sounds really good
2. I'm not quite sure. I'd rather add a new constructor for the API because on loading from disk the values are set and the setter methods invoke a network action which requires the world member not to be a nullptr. I did add a check if the object is a nullptr but you proposed to have a constructor which set's all values and not use the setter methods

**Comment by 12xx12** [comment]
Oh - do we have to update the plug-in with a or too? If yes I would do this also in this PR

**Comment by tigerw** [comment]
For 2. I added a 2nd constructor for normal spawning with a simplified param list. Which plugin do you mean?

**Comment by 12xx12** [comment]
The Core Plugin with the /summon command

**Comment by tigerw** [comment]
Good point, it would be nice to get the submodule updated all in one PR, but it's up to you.

**Review by tigerw — APPROVED** [review]

**Comment by 12xx12** [comment]
I'm gonna do a submodule update on another PR

**Pull request merged**